sopel/modules/wikipedia.py

```diff
@@ -321,26 +322,27 @@ def mw_image_description(server, image):
     return desc
 
 
-# Matches a wikipedia page (excluding spaces and #, but not /File: links), with a separate optional field for the section
-@plugin.url(r'https?:\/\/([a-z]+(?:\.m)?\.wikipedia\.org)\/wiki\/((?!File\:)[^ #]+)#?([^ ]*)')
+# Matches a Wikipedia link (excluding /File: pages)
+@plugin.url(r'https?:\/\/([a-z]+(?:\.m)?\.wikipedia\.org)\/wiki\/((?!File\:)[^ ]+)')
 @plugin.output_prefix(PLUGIN_OUTPUT_PREFIX)
 def mw_info(bot, trigger, match=None):
     """Retrieves and outputs a snippet of the linked page."""
     server = match.group(1)
-    query = unquote(match.group(2))
-    section = unquote(match.group(3))
+    page_info = urllib.parse.urlparse(match.group(2))
```

Comment:
To go with the other change.

```suggestion
    page_info = urlparse(match.group(2))
```